scoap3/utils/middleware.py

```diff
@@ -0,0 +1,35 @@
+from config.settings.production import MATOMO_SITE_ID, MATOMO_URL
+
+MATOMO_TRACKING_CODE = f"""
+
+<!-- Matomo -->
+<script>
+  var _paq = window._paq = window._paq || [];
+  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
+  _paq.push(['trackPageView']);
+  _paq.push(['enableLinkTracking']);
+  (function() {{
+    var u='{MATOMO_URL}';
+    _paq.push(['setTrackerUrl', u+'matomo.php']);
+    _paq.push(['setSiteId', '{MATOMO_SITE_ID}']);
+    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
+  }})();
+</script>
+<!-- End Matomo Code -->
+"""
+
+
+# TODO Do we need some cookie to be accepted for this?
+class MatomoMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        if response.get("Content-Type", "").startswith("text/html"):
+            response.content = response.content.decode("utf-8").replace(
+                "</head>", f"{MATOMO_TRACKING_CODE}</head>"
+            )
+            response.headers["Content-Length"] = len(response.content)
+        return response
```

Comment:
we could just add it here :)

https://github.com/SCOAP3/scoap3/blob/main/scoap3/templates/base.html#L4